src/Service/NullSearchService.php

```diff
@@ -0,0 +1,145 @@
+     */
+    public function index(ObjectManager $objectManager, $searchables, $requestOptions = [])
+    {
+        return new BatchIndexingResponse(['success' => true], 'foo');
+    }
+
+    /**
+     * @param object|array<int, object>                      $searchables
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return \Algolia\AlgoliaSearch\Response\AbstractResponse
+     */
+    public function remove(ObjectManager $objectManager, $searchables, $requestOptions = [])
+    {
+        return new BatchIndexingResponse(['success' => true], 'foo');
+    }
+
+    /**
+     * @param string                                         $className
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return \Algolia\AlgoliaSearch\Response\AbstractResponse
+     */
+    public function clear($className, $requestOptions = [])
+    {
+        return new NullResponse();
+    }
+
+    /**
+     * @param string                                         $className
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return \Algolia\AlgoliaSearch\Response\AbstractResponse
+     */
+    public function delete($className, $requestOptions = [])
+    {
+        return new NullResponse();
+    }
+
+    /**
+     * @param string                                         $className
+     * @param string                                         $query
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return array<int, object>
+     *
+     * @throws \Algolia\AlgoliaSearch\Exceptions\AlgoliaException
+     */
+    public function search(ObjectManager $objectManager, $className, $query = '', $requestOptions = [])
+    {
+        return [new \stdClass()];
+    }
+
+    /**
+     * @param string                                         $className
+     * @param string                                         $query
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return array<string, int|string|array>
+     *
+     * @throws \Algolia\AlgoliaSearch\Exceptions\AlgoliaException
+     */
+    public function rawSearch($className, $query = '', $requestOptions = [])
+    {
+        return [
+            'foo' => 'success',
+        ];
+    }
+
+    /**
+     * @param string                                         $className
+     * @param string                                         $query
+     * @param array<string, int|string|array>|RequestOptions $requestOptions
+     *
+     * @return int
+     *
+     * @throws \Algolia\AlgoliaSearch\Exceptions\AlgoliaException
+     */
+    public function count($className, $query = '', $requestOptions = [])
+    {
+        return 0;
+    }
+
+    /**
+     * @param object $entity
+     *
+     * @return bool
+     */
+    public function shouldBeIndexed($entity)
+    {
+        return true;
```

Comment:
This should return false.

Reply:
It would be better here to return real implementation, so tests can assert certain action results in indexing of certain entity.

Reply:
We aim to create a class that would just act as if there was no engine at all. I think it still can be a good starting point for a custom implementation with dummy responses